sympy/concrete/summations.py

```diff
@@ -1194,3 +1285,53 @@ def eval_sum_hyper(f, i_a_b):
                     return S.NegativeInfinity
             return None
         return Piecewise(res, (old_sum, True))
+
+
+def _eval_matrix_sum(expression):
+    f = expression.function
+    for n, limit in enumerate(expression.limits):
+        i, a, b = limit
+        dif = b - a
+        if dif.is_Integer:
+            if (dif < 0) == True:
+                a, b = b + 1, a - 1
+                f = -f
+
+            newf = eval_sum_direct(f, (i, a, b))
+            if newf is not None:
+                return newf.doit()
+
+
+def _dummy_with_inherited_properties_concrete(limits):
+    """
+    Return a Dummy symbol that inherits as much assumptions based on the
+    provided symbol and limits as possible.
+
+    If the symbol already has all possible assumptions, return None.
+    Also returns None if no limits are provided.
+    """
+    if len(limits) != 3:
+        return None
+    x, a, b = limits
+    l = (a, b)
+
+    assumptions_to_consider = ['extended_nonnegative', 'nonnegative',
+                               'extended_nonpositive', 'nonpositive',
+                               'extended_positive', 'positive',
+                               'extended_negative', 'negative',
+                               'integer', 'rational', 'finite',
+                               'zero', 'real', 'extended_real']
+
+    assumptions_to_keep = {}
+    assumptions_to_add = {}
+    for assum in assumptions_to_consider:
+        assum_true = x._assumptions.get(assum, None)
```

Comment:
Is there a reason to use`x._assumptions` rather than `getattr(x, 'is_' +assum)` here? Sometimes assumptions are just defined as class level attributes and I'm not sure that they would end up in `_assumptions`.

Reply:
I think the idea was just to pick the ones that were set and not derive any (as that can wait until they are needed), so this should be more efficient. Also, it was the first way I figured out...

I see your point, but have mainly (only?) tried with expressions where it is set. Note that this will always(?) be a symbol, so for this row I would not expect any issues. Maybe it makes more sense on the next line? Still, it seems like using e.g. integers give a correct answer, without really knowing how the code looks like.